### PR TITLE
Test against Ruby 3.3

### PR DESCRIPTION
**Pull request by tagliala**
Also update checkout version to v4

### CODE REVIEW

**Comment by tagliala** [comment]
Probably the following needs a fix too, because it is not considering latest Ruby versions

image

**Comment by grosser** [comment]
need to udpate nokogiri I think

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+version: 2
```
Comment (grosser) [inline .github/dependabot.yml:1]: don't really want that spam :)

Comment (tagliala) [reply .github/dependabot.yml:1]: sorry, you mean you prefer to remove dependabot?

Comment (grosser) [reply .github/dependabot.yml:1]: yeah, it's 9X% noise for projects like this, so I usually disable it

**Comment by grosser** [comment]
updated nokogiri here https://github.com/premailer/premailer/pull/441

**Comment by tagliala** [comment]
🤔 

I think this started failing again against nokogiri 1.16.0, let me check

**Comment by tagliala** [comment]
@grosser rebased and ready for review

In the settings -> branch protection, 2.7 should be removed as required workflow and 3.1, 3.2, 3.3 added. (Maybe 3.0 can be removed too)

**Comment by grosser** [comment]
nice :D

**Pull request merged**